src/stdatamodels/util.py

```diff
@@ -329,3 +338,14 @@ def rebuild_fits_rec_dtype(fits_rec):
         else:
             new_dtype.append((field_name, table_dtype))
     return np.dtype((np.record, new_dtype))
+
+
+def _fits_rec_to_array(fits_rec):
+    bad_columns = [n for n in fits_rec.dtype.fields if np.issubdtype(fits_rec[n].dtype, np.unsignedinteger)]
```

Comment:
I think so. To put it another way, the lack of unsigned integer support in FITS means that all unsigned integers in the fitsrec (so `field_dtype` is unsigned) will be stored as a signed integer with an offset (so `table_dtype` will be signed).

I'm happy to remove the first part of the check in `rebuild_fits_rec_dtype` if it would be preferable to have that change in this PR. Alternatively I can open a follow-up PR (or just a tracking issue if the optimization is of lower priority).

Reply:
I opened an issue to track optimizing the comparison.

https://github.com/spacetelescope/stdatamodels/issues/206

That way this PR can rely on the previously run regtests and a follow-up PR (after the next jwst release) can look into optimizing the comparison.